CMakeLists.txt

```diff
@@ -654,6 +654,21 @@ if (BUILD_TESTING)
 
     endforeach(test_case)
 
+    if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+        # s2n_connection_size_test - pointer size is different on 32 bit
+        #                            platforms resulting in different struct size
+        # other tests have 2038 time_t issues
+        set(CTEST_CUSTOM_TESTS_IGNORE
+            s2n_connection_size_test
+            s2n_cert_status_extension_test
+            s2n_self_talk_nonblocking_test
+            s2n_session_ticket_test
+            s2n_x509_validator_test
+        )
```

Comment:
This seems like a dangerous way to handle failures. Let's use your original in-code mechanism for skipping s2n_connection_size_test. For the others, we need an issue tracking that we need to fix them, because their failure means that our 32-bit build is broken.

Reply:
I don't have particularly strong opinions on this, but the decision was deliberate.

My preference is to not run the `s2n_connection_size` test at all on 32 bit platforms, rather than stubbing it out. The reason I prefer this is because the testing output feels a bit more semantically valid. E.g.
```
s2n_connection_size ................................ passed
```
should mean that we looked at the connection size and that it was good, not that we just skipped the test.

And it is absolutely intended to be a temporary workaround until #3903 is resolved. This was preferred over leaving the tests in a completely broken state, because my thinking was that this would at least allow us to unblock unit test runs in CI. This would be valuable because it could prevent any further breaking changes to the 32 bit build, even though we know that some of the functionality is already broken.

Basically, some unit tests are better than no unit tests.

The counter argument is that broken unit tests are very visible to end consumers, but I'm not sure that that is true because no one ever cut us an issue about these and they have been broken for quite a long time.

Reply:
Now that you've moved it to it's own test, it'll actually say "skipped" instead of "passed". Tests report "skipped" if you don't make any assertions in the test file.

I don't have a problem with this as a temporary fix for the other tests (as long as there's a specific issue tracking it-- we need to remember to remove them), but the connection test wouldn't be just temporarily ignored.

Reply:
Do you have a test that I can reproduce that on? I'm having trouble getting that behavior to happen
```
53: Test command: /home/ubuntu/workspace/s2n-tls/build/bin/s2n_connection_size_test
53: Environment variables: 
53:  S2N_DONT_MLOCK=1
53: Test timeout computed to be: 1500
53: Running /home/ubuntu/workspace/s2n-tls/tests/unit/s2n_connection_size_test.c ... I'm bailing out early bc 32 bit
1/1 Test #53: s2n_connection_size_test .........   Passed    0.00 sec

The following tests passed:
        s2n_connection_size_test

100% tests passed, 0 tests failed out of 1
```
I added a print statement to confirm that it's definitely bailing out early and that triggered as expected

I see the same behavior when running the default make target
```
222/233 Test #222: s2n_tls13_secrets_rfc8448_test ...................   Passed    0.01 sec
        Start  53: s2n_connection_size_test
223/233 Test   #4: s2n_aes_sha_composite_test .......................   Passed    0.01 sec
        Start 221: s2n_tls13_record_aead_test
224/233 Test  #53: s2n_connection_size_test .........................   Passed    0.01 sec
        Start 220: s2n_tls13_prf_test
225/233 Test #220: s2n_tls13_prf_test ...............................   Passed    0.01 sec
```

Reply:
Here's the code that does the printing: https://github.com/aws/s2n-tls/blob/main/tests/s2n_test.h#L42-L58

You should be able to find some skipped tests if you look at a unit test run using openssl-1.0.2. We skip a lot for that libcrypto :)

Reply:
Had a quick chat offline. Unfortunately the macro won't help us since cmake zealously swallows all output, but the s2n_connection_size test will still be moved to use runtime filtering since that is what we are currently doing in our other testing.